layers/+web/eww/funcs.el

```diff
@@ -45,9 +53,25 @@
   "Return a condensed/full hint for the eww transient state"
   (concat
    " "
-   (if spacemacs--eww-ts-full-hint-toggle
-       spacemacs--eww-ts-full-hint
-     (concat "[" (propertize "?" 'face 'hydra-face-red) "] help"))))
+   (if (not spacemacs--eww-ts-full-hint-toggle)
+       (concat "[" (propertize "?" 'face 'hydra-face-red) "] help")
+     (let ((result spacemacs--eww-ts-full-hint))
+       (cl-loop for (flag . reg) in
+                '((spacemacs--eww-ts-use-evil . "\\[[^\\[]+ scroll [^\\[]*")
+                  (spacemacs--eww-ts-use-writeroom . "\\[[^\\[]+writeroom[^\\[]*")
+                  (spacemacs--eww-ts-use-zoom-frm . "\\[[^\\[]+zoom-in[^\\[]*")
+                  (spacemacs--eww-ts-use-zoom-frm . "\\[[^\\[]+zoom-out[^\\[]*")
+                  (spacemacs--eww-ts-use-zoom-frm . "\\[[^\\[]+unzoom[^\\[]*"))
+                do
+                (when-let* (((not flag))
+                            ((string-match reg result))
```

Comment:
I don't think this works as expected, you need to check the value of the variable whose name is stored in `flag`, not the value of `flag` itself.  Either use `symbol-value` or don't quote the symbols in the iterated list.